astrapy/db.py

```diff
@@ -694,6 +697,51 @@ def insert_many(
 
         return response
 
+    def batched_concurrent_insert_many(
+        self,
+        documents: List[API_DOC],
+        options: Optional[Dict[str, Any]] = None,
+        partial_failures_allowed: bool = False,
+        batch_size: int = MAX_INSERT_BATCH_SIZE,
+        concurrency: int = 1,
+    ) -> List[API_RESPONSE]:
+        """
+        Batch insert multiple documents into the collection with concurrency.
+        Args:
+            documents (list): A list of documents to insert.
+            options (dict, optional): Additional options for the insert operation.
+            partial_failures_allowed (bool, optional): Whether to allow partial failures in the batch.
+            batch_size (int, optional): Override the default insertion batch size.
+            concurrency (int, optional): The number of concurrent batch insertions.
+        Returns:
+            list: The responses from the database after the batched insert operation.
+        """
+
+        # Split the documents into batches
+        batched_list = [
+            documents[i : i + batch_size] for i in range(0, len(documents), batch_size)
+        ]
+
+        # Function to insert a single batch
+        def insert_batch(batch: List[API_DOC], index: int) -> API_RESPONSE:
```

Comment:
Can't we use `self.insert_many` directly ? It's a lot of code just to add a debug log